### PR TITLE
new tutorial for a matrix homeserver setup on a hetzner cloud server

### PR DESCRIPTION
**Pull request by k6wcQDfv**

### CODE REVIEW

File: tutorials/install-and-configure-matrix/01.en.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    volumes:
+      - ./data/traefik/letsencrypt:/letsencrypt
+      - "/var/run/docker.sock:/var/run/docker.sock:ro"
+    command:
+      #- "--log.level=DEBUG"
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--providers.docker.exposedbydefault=false"
+      - "--entrypoints.websecure.address=:443"
+      - "--entrypoints.synapse.address=:8448"
```
Comment (rook59) [inline tutorials/install-and-configure-matrix/01.en.md:115]: Why is this required? In your SRV you point to port 443 only

File: tutorials/install-and-configure-matrix/01.en.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+      - "traefik.http.routers.synapse.rule=PathPrefix(`/_matrix{regex:$$|/.*}`)"
+      - "traefik.http.routers.synapse.tls.certresolver=myresolver"
+    volumes:
+      - ./data/synapse:/data
+    restart: always
+    depends_on:
+      - "postgres"
+
+  postgres:
+    image: postgres:latest
```
Comment (rook59) [inline tutorials/install-and-configure-matrix/01.en.md:92]: Can you pin a postgres major version? Otherwise this might break sooner or later

File: tutorials/install-and-configure-matrix/01.en.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+      - POSTGRES_DB=synapse
+      - POSTGRES_INITDB_ARGS="-E UTF-8"
+      - LC_COLLATE=C
+      - LC_CTYPE=C
+    volumes:
+      - ./data/postgres:/var/lib/postgresql/data
+    restart: always
+
+  traefik:
+    image: traefik:latest
```
Comment (rook59) [inline tutorials/install-and-configure-matrix/01.en.md:105]: Can you pin a traefik major version? Otherwise this might break sooner or later

File: tutorials/install-and-configure-matrix/01.en.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+- selfhosted federation
+- ...
+
+The community at `matrix.org` is always a good start in order to extend your knowledge and customize your homeserver according to your needs.
+
+
+
+
+
+
```
Comment (rook59) [inline tutorials/install-and-configure-matrix/01.en.md:226]: Please re-add MIT license block from the template

File: tutorials/install-and-configure-matrix/01.en.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+## Step 3.3 Login and test
+
+The general availability of your new matrix homeserver can be tested via:
+
+`https://sub.example.com/_matrix/static/`
+
+You should see a landing page and follow the link to the clients page in order to choose and install a client.
+
+## Conclusion
```
Comment (rook59) [inline tutorials/install-and-configure-matrix/01.en.md:211]: Maybe also mention https://federationtester.matrix.org/ here to test the matrix server?

File: tutorials/install-and-configure-matrix/01.en.md (hunk 1 above)
Comment (k6wcQDfv) [reply tutorials/install-and-configure-matrix/01.en.md:115]: 8448 is the known default port for a matrix api exposure. A client may always lookup this port, not every software out there uses a srv record in order to find out where to connect. Thus in my opinion it's good practice to always expose 8448 for compatiblity.

File: tutorials/install-and-configure-matrix/01.en.md (hunk 2 above)
Comment (k6wcQDfv) [reply tutorials/install-and-configure-matrix/01.en.md:92]: > Otherwise this might break sooner or later
Which is exactly what I intend: Anything connected to the internet needs continuous maintenance and monitoring. If a setup is running in the same configuration unmaintained for several years, chances are high it's not your system anymore and you've successfully exposed your users data and privacy to someone else. This should be of major concern especially for this kind of software.

Since this tutorial might be available for several years including the fact that there might be no continuous review for it's actuality it should be designed to break in case a review is needed, not silently add up to the pile of burden and botnets within our networks. Therefore it is written the way it is.
